### PR TITLE
Rewrite in Go: Build in Docker

### PR DESCRIPTION
**Pull request by dokshina**

### CODE REVIEW

File: src/commands/pack.go [hunk 1]
```diff
@@ -85,7 +95,25 @@ instantiated unit file
 Used for rpm and deb types
 `
 
-	stateboardUnitTemplateFlagDoc = `Path to the template for stateboard systemd unit file
+	stateboardUnitTemplateFlagDoc = `Path to the template for
+stateboard systemd unit file
 Used for rpm and deb types
+`
+
+	useDockerDoc = `forces to build the application in Docker`
```
Comment (olegrok) [inline src/commands/pack.go:103]: `forces ` -> `Forces `

File: src/commands/pack.go [hunk 2]
```diff
@@ -85,7 +95,25 @@ instantiated unit file
...
+Used on build in docker
+Default to Dockerfile.build.cartridge
+`
+
+	sdkLocalDoc = `SDK from the local machine should be
+delivered in the result artifact
+Used for building in docker with Tarantool Enterprise
+`
+
+	sdkPathDoc = `path to the SDK to be delivered in the result artifact
```
Comment (olegrok) [inline src/commands/pack.go:115]: `path ` -> `Path `

File: src/common/utils.go [hunk 3]
```diff
@@ -308,3 +307,36 @@ func GetNextMajorVersion(version string) (string, error) {
 
 	return strconv.Itoa(major + 1), nil
 }
+
+func GetTarantoolRepoVersion(version string) string {
```
Comment (olegrok) [inline src/common/utils.go:311]: Does it consider that 2.1 is 2_x? However check it please, may be it has been already fixed.

File: src/docker/run.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	WorkingDir string
+	Cmd        []string
+
+	Volumes map[string]string
+
+	Quiet bool
+	Debug bool
+}
+
+func waitStartOutput(out io.ReadCloser, quiet bool) error {
```
Comment (olegrok) [inline src/docker/run.go:31]: I think you should provide an interface that get function and flag and enable spinner if flag is true.

It's not good when you meet waitgroups/channels creations to run spinner over all project.

`run_in_background(myFunction, enableSpinner)`.

File: src/docker/run.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	defer func() {
+		if opts.Debug {
+			log.Warnf("Build container %s is not removed due to debug mode", containerID)
+			return
+		}
+
+		log.Infof("Remove build container...")
+		err := cli.ContainerRemove(ctx, containerID, types.ContainerRemoveOptions{
+			// RemoveVolumes: true,
```
Comment (olegrok) [inline src/docker/run.go:101]: drop it if unused

File: src/docker/run.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+		Cmd:        opts.Cmd,
+		WorkingDir: opts.WorkingDir,
+		Tty:        true,
+	}
+
+	hostConfig := container.HostConfig{
+		Binds: binds,
+	}
+
+	// create container
```
Comment (olegrok) [inline src/docker/run.go:85]: Too obvious for "ContainerCreate" function

File: src/docker/run.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+		}
+	}()
+
+	// start container
+	if err := cli.ContainerStart(ctx, containerID, types.ContainerStartOptions{}); err != nil {
+		return fmt.Errorf("Failed to start container: %s", err)
+	}
+
+	out, err := cli.ContainerLogs(ctx, containerID, types.ContainerLogsOptions{
+		ShowStdout: true,
```
Comment (olegrok) [inline src/docker/run.go:114]: Does it respect "quiet" mode?

File: src/docker/run.go [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+	}
+
+	if err := waitStartOutput(out, opts.Quiet); err != nil {
+		return err
+	}
+
+	if code, err := cli.ContainerWait(ctx, containerID); err != nil {
+		return fmt.Errorf("Failed to run container: %s", err)
+	} else if code != 0 {
+		return fmt.Errorf("Failed to run command on container")
```
Comment (olegrok) [inline src/docker/run.go:128]: Put error code here please

Comment (olegrok) [reply src/docker/run.go:128]: If it's internal "docker" error code

File: src/project/project.go [hunk 9]
```diff
@@ -24,17 +24,25 @@ type ProjectCtx struct {
...
 	TarantoolVersion      string
 	TarantoolIsEnterprise bool
 	WithStateboard        bool
 
+	BuildInDocker       bool
+	BuildFrom           string
+	BuildDockerfileBase string
+	SDKLocal            bool
+	SDKPath             string
+	BuildSDKDirame      string
```
Comment (olegrok) [inline src/project/project.go:44]: `BuildSDKDirame ` -> `BuildSDKDirname`?

File: src/docker/run.go (hunk 7 above)
Comment (dokshina) [reply src/docker/run.go:114]: Yes, it collects all logs, but shows them only when quiet mode isn't enabled.

File: src/docker/run.go (hunk 4 above)
Comment (dokshina) [reply src/docker/run.go:31]: Yes, I sure that all `spinner` magic should be placed in some common module.

Are you OK with doing it in a separated PR? I've added it to checklist of #242

**Comment by olegrok** [comment]
Of course

File: src/docker/run.go (hunk 4 above)
Comment (olegrok) [reply src/docker/run.go:31]: Of course

File: src/common/utils.go (hunk 3 above)
Comment (dokshina) [reply src/common/utils.go:311]: Fixed to replace `2_1` to `2x`.

**Pull request merged**